<launch>
    <!-- Set the time to be the one of the simulation -->
    <param name="/use_sim_time" value="true" />

    <!-- Start odom to tf converter node with parameters-->
    <param name="root_frame" type="string" value="odom" />
    <param name="child_frame" type="string" value="base_link" />
    <node pkg="second_project" type="ugv_odom_to_tf" name="ugv_odom_to_tf" output="screen">  </node>

    <!-- run pointcloud_to_laserscan node -->
        <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan">
            <remap from="cloud_in" to="/ugv/rslidar_points"/>
            <rosparam>
                target_frame: rslidar
                transform_tolerance: 0.01
                min_height: 0.0
                max_height: 1.0

                angle_min: -3.14 # -M_PI
                angle_max: 3.14 # M_PI
                angle_increment: 0.0087 # M_PI/360.0
                scan_time: 0.3333
                range_min: 0.45
                range_max: 10.0
                use_inf: true

                # Concurrency level, affects number of pointclouds queued for processing and number of threads used
                # 0 : Detect number of cores
                # 1 : Single threaded
                # 2->inf : Parallelism level
                concurrency_level: 0
            </rosparam>
        </node>

        <node pkg="tf" type="static_transform_publisher" name="laser_tf" args="0 0 0 0 0 0 1 base_link rslidar 100" />

    <!-- Start gmapping with parameters-->
    <node pkg="gmapping" type="slam_gmapping" name="gmapping" output="screen">
      <!--<remap from="scan" to="base_scan" />
      <param name="xmax" value="5"/>
      <param name="ymax" value="5"/>
      <param name="xmin" value="-5"/>
      <param name="ymin" value="-5"/>
      <param name="delta" value="0.05"/>-->
    </node>


    <node name="rviz" pkg="rviz" type="rviz" args="-f map -d $(find second_project)/cfg/rviz_config.rviz"/>

    <node name="rqt_tf_tree" pkg="rqt_tf_tree" type="rqt_tf_tree" />
</launch>
